Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -968,6 +972,40 @@ bool vtkMRMLSliceLogic::UpdateFractions(vtkImageMathematics* fraction, double op
   return modified;
 }
 
+//----------------------------------------------------------------------------
+void vtkMRMLSliceLogic::UpdateReconstructionSlab(vtkMRMLSliceLogic* sliceLogic, vtkMRMLSliceLayerLogic* sliceLayerLogic)
+{
+  if (!sliceLogic || !sliceLayerLogic || !sliceLogic->GetSliceNode())
+    {
+    return;
+    }
+
+  vtkImageReslice* reslice = sliceLayerLogic->GetReslice();
+  vtkMRMLSliceNode* sliceNode = sliceLayerLogic->GetSliceNode();
+
+  double sliceSpacing;
+  if (sliceNode->GetSliceSpacingMode() == vtkMRMLSliceNode::AutomaticSliceSpacingMode)
+    {
+    sliceSpacing = *sliceLogic->GetLowestVolumeSliceSpacing();
+    }
+  else
+    {
+    sliceSpacing = sliceNode->GetPrescribedSliceSpacing()[2];
+    }
+  sliceSpacing = sliceSpacing ? sliceSpacing : sliceLogic->DefaultSlabReconstructionThickness;
```

Comment:
Since `GetLowestVolumeSliceSpacing()` already take care of checking if spacing mode is prescribed or automatic, we should be able to simplify this.

I am not clear why in one case we get the first spacing value and in the other case, the last one.

Reply:
I believe I misunderstood this [comment](https://github.com/stephencrowell/Slicer/blob/59ce1e2786df4b97e091d31a8adc3a0e78a9b2fc/Libs/MRML/Core/vtkMRMLSliceNode.h#L408) about how to get slice spacing.